src/events.rs

```diff
@@ -83,7 +88,19 @@ impl MuxedEvents {
             || self.watched_directories.contains_key(&path.to_path_buf())
     }
 
-    fn add_directory(&mut self, path: impl AsRef<Path> + Into<PathBuf>) -> Result<(), Error> {
+    fn watch(watcher: &mut notify::RecommendedWatcher, path: impl AsRef<Path>) -> io::Result<()> {
+        use notify::Watcher;
+        watcher
+            .watch(path.as_ref(), notify::RecursiveMode::NonRecursive)
+            .map_err(notify_to_io_error)
+    }
+
+    fn unwatch(watcher: &mut notify::RecommendedWatcher, path: impl AsRef<Path>) -> io::Result<()> {
+        use notify::Watcher;
+        watcher.unwatch(path.as_ref()).map_err(notify_to_io_error)
+    }
+
+    fn add_directory(&mut self, path: impl AsRef<Path> + Into<PathBuf>) -> io::Result<()> {
```

Comment:
Instead of specifying `Into<PathBuf>` here, you can use `path.into_path_buf()` to replace `path.into()` below. `into_path_buf()` is already available for the `Path` type.

Reply:
You mean [`to_path_buf`](https://doc.rust-lang.org/std/path/struct.Path.html#method.to_path_buf) right? My thinking was that if you already had a `PathBuf` passed in, the `path.into()` would be a no-op instead of a re-allocation from the reference type. Looks like even tokio generally opts for the re-alloc though, so I'll just do that as well.

Ref: https://github.com/tokio-rs/tokio/blob/d65bf3805be376505f5c2fb0724cec8917dfb813/tokio/src/fs/file.rs#L117-L118

Reply:
> You mean to_path_buf right?

Yeah 🤦‍♂ 

> My thinking was that if you already had a PathBuf passed in, the path.into() would be a no-op instead of a re-allocation from the reference type.

That's a really good point that I hadn't thought of! I don't think one extra realloc is going to break the bank here, but the fact that it will realloc is good to keep in mind.